ruff.toml

```
@@ -0,0 +1,29 @@
+preview = true
+
+extend-exclude = []
+
+line-length = 100
+indent-width = 4
+
+# Assume Python 3.11
+target-version = "py311"
+
+[lint]
+# Here we set the rules we want to use.
+# E1 - spacing/identation (https://docs.astral.sh/ruff/rules/#pycodestyle-e-w)
+# E22 - whitespace (https://docs.astral.sh/ruff/rules/#pycodestyle-e-w)
+# F - (https://docs.astral.sh/ruff/rules/#pyflakes-f)
+# C901 - complex-structure (https://docs.astral.sh/ruff/rules/#pycodestyle-e-w)
+# N - naming convention (https://docs.astral.sh/ruff/rules/#pep8-naming-n)
+# COM - commas (https://docs.astral.sh/ruff/rules/#flake8-commas-com)
+# I - isort (https://docs.astral.sh/ruff/rules/#isort-i)
+# We could also add the following rules:
+# "ANN0" - missing-type-function-argument (https://docs.astral.sh/ruff/rules/#flake8-annotations-ann)
+# "ANN201" - missing-return-type-undocumented-public-function (https://docs.astral.sh/ruff/rules/#flake8-annotations-ann)
+# Sometimes useful:
+# ERA - commented-out-code (https://docs.astral.sh/ruff/rules/#eradicate-era)
+select = ["E1", "E22", "E252", "E4", "E7", "E9", "F", "W291", "C901", "N", "COM818", "COM819", "I"]
+# Here we set the rules we want to ignore.
+# N804 - invalid-first-argument-name-for-class-method (https://docs.astral.sh/ruff/rules/#pep8-naming-n)
+# F821 - undefined-name (already handled by VSCode) (https://docs.astral.sh/ruff/rules/#pyflakes-f)
+ignore = ["N804", "F821"]
```

Can we not ignore F821? Since I like that it gives an actual error rather than a warning for this due to Ruff

```suggestion
ignore = ["N804"]
```

---

Sure